### DISCUSSION TITLE
Recursive Component

### DISCUSSION THREAD

**Discussion by lucadegasperi** [livewire/livewire#2170] [1. Help]
Hello,

I'm trying to build an infinite pagination by nesting a Livewire component inside itself but it seems the nesting goes maximum to 1 level deep. Is what I'm wanting to do correct or is there another way to do it?

This is the list-entries component
```
@foreach($entries as $entry)
    // here I display the fetched entries
@endforeach
@if(count($entries) == 0)
    <a wire:click.prevent="fetch">Fetch More</a>
@else
    <livewire:list-entries :page="($page + 1)" />
@endif
```

**Comment by joshhanley** [comment]
@lucadegasperi instead of using pagination, why don't you just have a property on your component class called `$numberOfResults = 10` or something like that. If you were to show 10 at a time, you could just have a method that adds an extra 10 `$this->numberOfResults += 10;` each time "Fetch More" is clicked.

Then just show results based off this limit instead of using pagination.

Hope this helps!

**Comment by lucadegasperi** [comment]
@joshhanley since I'm new to livewire, maybe I misunderstand how it works, won't your solution cause the component to re-render the whole view and send it over the wire, every time the "Fetch More" button is pressed?

At first you have 10 results and you show them, then you press "Fetch More", you append the next 10 results to the array of results and send the whole 20 results html over the wire. Or is livewire smart enough to render only the new added results?

Thank you for your time

**Comment by joshhanley** [comment]
@lucadegasperi ah yeah, you are right, it would re-fetch all results. Hmm there was an open issue about it #590 but no good API solution was proposed so Caleb put it on the back burner. Found a few other threads with similar issues #1225 #1289 #1757. 

So an idea could be that you have a child component, who takes an offset and a limit as properties, and handles fetching records based on that (rather than use pagination).
And then have a parent component who keeps track of the total, controls the number of child components on the page, and has the show more button which would add another child component with a new offset.

In theory once the child components are initialised on the page (and you have wire:key setup properly) then the child components shouldn't refresh when the parent does, therefore should only get the list of data for its subset of data once.

And the only data the parent should be transmitting is child component ids and the total.

I don't know definitely if that's the case but it would be worth experimenting.

I'm happy to help you to get an experiment working as I might have need of something similar soon.

Let me know what you think!
